### PR TITLE
perl-xsloader: add new package

### PR DESCRIPTION
**Pull request by ChristopherChristofi**

### CODE REVIEW

**Review by alecbcs — DISMISSED** [review]
Confirmed SHA. Looks good to me. @ChristopherChristofi thanks for the new package.

File: var/spack/repos/builtin/packages/perl-xsloader/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/perl-xsloader/package.py]:
```suggestion
# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
```

@ChristopherChristofi not required but if you could update the license for 2024 that'd be great.

**Comment by EbiArnie** [comment]
XSLoader is a core Perl module and should not become a separate spack package, in my opinion.
See here: https://perldoc.perl.org/modules

**Comment by alecbcs** [comment]
@ChristopherChristofi any reason in particular you'd like to have `xsloader` added as an additional package outside of the core Perl install?

**Comment by ChristopherChristofi** [comment]
Thank you both.
@alecbcs I suspect pointing to the core perl module would be fine for producing successful builds where dependent packages are concerned, so I do not think it is an issue.
I assumed it would be good to make available all possible requirements for a spack package in its recipe, so they're accounted for. Although not exactly the same, similar to how we always want to avoid hidden vendored dependencies.
For ones @EbiArnie has marked (#41910, #41911), which all make sense, have their own distribution in metacpan and don't fall under something like: https://metacpan.org/dist/perl, otherwise I would've not added it.
Plus, for whatever reason, where it is not a dependency, an end-user might want: `module load perl-xsloader` available to them

**Comment by EbiArnie** [comment]
I don't have a strong opinion to do it one way or another.

So far, Spack packages for core modules (broadly) don't exist.
I assumed this was the policy.

One could probably create virtuals for the Perl modules that come with core Perl and have the Spack 'perl' package provide those.
The advantage would be that these are versioned. If I have to depend e.g. on a new XSLoader, that would automatically turn into a dependency on a more recent version of Perl.
Also, Perl modules can become part of core or can be removed. Test2 is a more recent example for this, since v5.25.1.
See here for reference: [corelist](https://metacpan.org/release/BINGOS/Module-CoreList-3.02/view/corelist)
So another advantage is that this approach is more correct in modelling the actual dependencies.

The disadvantage is that we have more nodes for the solver and a lot of packages that are just 'perl' under the hood.

Thoughts?

As for the `module load perl-xsloader` - what is the advantage / difference to just `module load perl` ?

**Comment by ChristopherChristofi** [comment]
Thanks again for your feedback. I see the main advantage as something similar to how you described it with your Test2 / corelist examples. By ensuring the continued availability of dependency solutions that might involve standard Perl modules which may change or be removed.
I didn’t plan on adding all those listed on the standard list, primarily ones I have come across used and included as dependencies for various potential non-core Perl module packages. Hence why I also mention above that I guess not adding them in the recipe is fine for successful builds alone, 
although such recipes would not reflect the complete dependency requirements, and perhaps become faulty if it has an unaccounted for dependency that is no longer a core Perl module.

**Comment by ChristopherChristofi** [comment]
```
Data for 2023-07-02
XSLoader was first released with perl v5.6.0
```
Has been a core module since 5.6.0 (release date 2000), which pre-dates all end-of-life Perl releases in spack

**Pull request closed without merging**